src/sentry/incidents/models.py

```diff
@@ -458,6 +458,11 @@ class Meta:
 
     __repr__ = sane_repr("id", "name", "date_added")
 
+    def save(self, **kwargs: Any) -> None:
+        if self.owner_id is not None and self.team_id is None and self.user_id is None:
+            raise ValueError("AlertRule with owner requires either team_id or user_id")
+        return super().save(**kwargs)
```

Comment:
If we want to uphold this condition, should we add an overridden update method that checks this as well?

Reply:
👍 will add